### ISSUE TITLE
Support for JSON Merge Patch (RFC 7386)

### ISSUE THREAD

**Issue by Tiberriver256** [dotnet/aspnetcore#35237]
### Is your feature request related to a problem? Please describe.
Implementing JSON Merge Patch manually can be quite tricky but provides a simpler experience for API consumers.

### Describe the solution you'd like
Currently there is built-in support for JSON Patch ([RFC 6902](https://datatracker.ietf.org/doc/html/rfc6902)) via [JsonPatchDocument](https://docs.microsoft.com/en-us/aspnet/core/web-api/jsonpatch?view=aspnetcore-5.0). Would love to also see an implementation of  [RFC 7386](https://datatracker.ietf.org/doc/html/rfc7386).

**So no one has to wade through an RFC...**

Given the following JSON representation:

```json
{
  "name": "Joe",
  "email": "joe@example.com",
  "physicalAttributes": { "weight": 75, "height": 175 },
  "favoriteColors": ["blue", "red"]
}
```

These two will produce equivalent modifications:

#### JSON Patch

```json
[
   { "op": "remove", "path": "/email" },
   { "op": "add", "path": "/favoriteColors/-", "value": "black" }
]
```

#### JSON Merge Patch

```json
{
  "email": null,
  "favoriteColors": ["blue", "red", "black"]
}
```

### Additional context
OData has an implementation of this spec via [Delta<T>](https://github.com/OData/AspNetCoreOData/blob/ea6775e7c75c3d198bf86e041bcce27ad4f457db/src/Microsoft.AspNetCore.OData/Deltas/Delta.cs) but it [was not designed](https://stackoverflow.com/a/14734273/3317144) to work with formatters other than OData.

Here are a couple of open-source implementations:
https://www.strathweb.com/2013/01/easy-asp-net-web-api-resource-updates-with-delta/
https://github.com/Morcatko/Morcatko.AspNetCore.JsonMergePatch

**Comment by mitsha-microsoft** [comment]
Hi,
Our service uses PATCH primarily for a huge set of our APIs and having something built-in that handles RFC 7386 will help us greatly (otherwise we will need to roll out our own implementation or use [a third party OSS](https://github.com/Morcatko/Morcatko.AspNetCore.JsonMergePatch) package).

**Comment by desjoerd** [comment]
@mikekistler I totally agree that this is the more friendly patch method like you mentioned in the .NET 10 Preview4 unboxed!.

I currently use my "own" wrapper value to be able to do this https://github.com/desjoerd/OptionalValues. But having native support for this would be awesome.

I also have an horrible implementation which runs on json level... https://github.com/desjoerd/toolbox/tree/main/dotnet/Toolbox/JsonMerge

Is adding this something which is open for help?

**Comment by SemaphoreSlim1** [comment]
DotNext already has this implemented, tested, and ready to go - and has the benefit of being "official" Microsoft code.
https://dotnet.github.io/dotNext/features/core/optional.html
